lrpar/src/lib/ctbuilder.rs

```diff
@@ -754,57 +753,68 @@ where
-        #![allow(clippy::unnecessary_wraps)]
-        #![deny(unsafe_code)]
-        #[allow(unused_imports)]
-        use super::*;
-",
-        );
+            Some(str::parse::<TokenStream>(&self.gen_user_actions(grm)?).unwrap())
+        } else {
+            None
+        };
 
-        outs.push_str(&self.gen_parse_function(grm, stable)?);
-        outs.push_str(&self.gen_rule_consts(grm));
-        outs.push_str(&self.gen_token_epp(grm));
-        match self.yacckind.unwrap() {
+        let rule_consts = str::parse::<TokenStream>(&self.gen_rule_consts(grm)).unwrap();
+        let token_epp = str::parse::<TokenStream>(&self.gen_token_epp(grm)).unwrap();
+        let parse_function = self.gen_parse_function(grm, stable)?;
+        let action_wrappers = match self.yacckind.unwrap() {
             YaccKind::Original(YaccOriginalActionKind::UserAction) | YaccKind::Grmtools => {
-                outs.push_str(&self.gen_wrappers(grm));
+                Some(str::parse::<TokenStream>(&self.gen_wrappers(grm)).unwrap())
             }
             YaccKind::Original(YaccOriginalActionKind::NoAction)
-            | YaccKind::Original(YaccOriginalActionKind::GenericParseTree) => (),
+            | YaccKind::Original(YaccOriginalActionKind::GenericParseTree) => None,
             _ => unreachable!(),
-        }
-        outs.push_str("    } // End of `mod _parser_`\n\n");
-        outs.push_str("    #[allow(unused_imports)]\n");
-        outs.push_str("    pub use _parser_::*;\n");
-        outs.push_str("    #[allow(unused_imports)]\n");
-        outs.push_str("    use ::lrpar::Lexeme;\n");
-
-        // Output the cache so that we can check whether the IDs map is stable.
-        outs.push_str(cache);
-        outs.push_str("} // End of `mod {mod_name}` \n\n");
+        };
+        let mod_name = format_ident!("{}", mod_name);
+        let out_tokens = quote! {
+            #visibility mod #mod_name {
+                #cache
+                // gen_user_actions
+                #user_actions
+
+                mod _parser_ {
+                    #![allow(clippy::type_complexity)]
+                    #![allow(clippy::unnecessary_wraps)]
+                    #![deny(unsafe_code)]
+                    #[allow(unused_imports)]
+                    use super::*;
+                } // End of `mod _parser_`
+
+                #parse_function
+                #rule_consts
+                #token_epp
+                #action_wrappers
+
+                #[allow(unused_imports)]
+                pub use _parser_::*;
+                #[allow(unused_imports)]
+                use ::lrpar::Lexeme;
+            } // End of `mod {#mod_name}`
+        };
 
         let mut f = File::create(outp_rs)?;
+        let syntax_tree = syn::parse_str(&out_tokens.to_string()).unwrap();
+        // This unfortunately reacts pretty badly to the consts, generating many lines
+        let outs = prettyplease::unparse(&syntax_tree);
```

Comment:
Filed https://github.com/dtolnay/prettyplease/issues/105 regarding this comment.

Reply:
fixed via [5ce16e3](https://github.com/softdevteam/grmtools/pull/510/commits/5ce16e3613798f284d62b89a06afa541e8c5e649)

Reply:
Wow that was quick!